db/view/view.cc

```diff
@@ -1109,102 +1124,65 @@ future<> mutate_MV(
+        auto my_address = utils::fb_utilities::get_broadcast_address();
+        auto remote_it = std::find(remote_endpoints.begin(), remote_endpoints.end(), my_address);
+        if (remote_it != remote_endpoints.end()) {
+            if (!target_endpoint) {
+                target_endpoint = *remote_it;
+                remote_endpoints.erase(remote_it);
             } else {
-                vlogger.debug("Sending view update to endpoint {}, with pending endpoints = {}", *paired_endpoint, pending_endpoints);
-                // Note we don't wait for the asynchronous operation to complete
-                // without a batchlog, and without checking for success.
-                // When the ownership of the view partition is being moved to a
-                // new node (or nodes), listed in pending_enpoints, we also need
-                // to send the update there. Currently, we do this from *each* of
-                // the base replicas, but this is probably excessive - see
-                // See https://issues.apache.org/jira/browse/CASSANDRA-14262/
-                tracing::trace(tr_state, "Sending view update for {}.{} to {}, with pending endpoints = {}; base token = {}; view token = {}",
-                        mut.s->ks_name(), mut.s->cf_name(), *paired_endpoint, pending_endpoints, base_token, view_token);
-                future<> view_update = service::get_local_storage_proxy().send_to_endpoint(
-                        std::move(mut),
-                        *paired_endpoint,
-                        std::move(pending_endpoints),
-                        db::write_type::VIEW,
-                        std::move(tr_state),
-                        stats,
-                        allow_hints).then_wrapped(
-                                [paired_endpoint,
-                                 is_endpoint_local,
-                                 updates_pushed_remote,
-                                 maybe_account_failure = std::move(maybe_account_failure)] (future<>&& f) mutable {
-                    return maybe_account_failure(std::move(f), std::move(*paired_endpoint), is_endpoint_local, updates_pushed_remote);
-                });
-                if (wait_for_all) {
-                    fs->push_back(std::move(view_update));
+                // Remove the duplicated entry
+                if (*target_endpoint == *remote_it) {
+                    remote_endpoints.erase(remote_it);
                 } else {
-                    // The update is sent to background in order to preserve availability,
-                    // its parallelism is limited by view_update_concurrency_semaphore
-                    (void)view_update;
+                    std::swap(*target_endpoint, *remote_it);
                 }
             }
-        } else if (!pending_endpoints.empty()) {
-            // If there is no paired endpoint, it means there's a range movement going on (decommission or move),
```

Comment:
We lost this comment in this change. I'm not sure it's a big loss - I don't remember what it actually means and can't understand it now :-)